### DISCUSSION TITLE
Publish with Github Pages Keeping repo as Private on free plan

### DISCUSSION THREAD

**Discussion by shemeerp** [community/community#22817] [Other Feature Feedback, Questions, & Ideas]
<p>Hi,</p>
<p>I have a free plan. I have to host site with github pages. I have purchased a theme, so I need to make the code/repo private. Can I host on Github pages keeping the repo as private ? My Site will be available publicly</p>
<p>Thanks,</p>
<p>Shemeer</p>

**Comment by Joness666** [comment]
<p>i think yes, because you pay money</p>

**Comment by RDIL** [comment, marked as answer]
<p>No, to use <a href="https://pages.github.com" rel="nofollow noopener">GitHub pages</a> on a private repository, you need the <a href="https://github.com/pricing" rel="nofollow noopener">PRO</a> plan.</p>

**Reply by javiyadeep** [reply]
i have pro plan but i can't do host

**Reply by blakete** [reply]
@RDIL Can you do it with the Team plan or only on the Enterprise plan?

**Comment by makerofgames** [comment]
<blockquote>
<p><a href="https://begamedev.com" rel="nofollow noopener">@shemeerp</a> wrote:</p>
</blockquote>
<blockquote>
<p>Hi,</p>
</blockquote>
<blockquote>
<p>I have a free plan. I have to host site with github pages. I have purchased a theme, so I need to make the code/repo private. Can I host on Github pages keeping the repo as private ? My Site will be available publicly</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Thanks,</p>
</blockquote>
<blockquote>
<p>Shemeer</p>
</blockquote>
<p>Hey friend, if you dont find your answer in github then i wanted to say that there are free hostings (even fully working ones without ads) or very cheap ones not exceeding few dollars yearly.</p>
<p>Cheers</p>

**Comment by nyahbhinghiPrincess** [comment]
<p>Hi <a class="mention" href="/fairol85">@fairol85</a>,</p>
<p>Thank you for reaching out in this thread!   You are more likely to get a useful response if you are explicit or give a few more details in your question. Please let us know what information you are looking for or asking about.</p>

**Comment by KyleStay** [comment]
<p>I have a PRO plan, but I’m still getting this message in my repo. Has something changed?</p>
<p><img src="https://user-images.githubusercontent.com/4483057/181073224-11dac69d-a602-44ce-a78b-007365c803dc.png" alt="Snag_2fff5a.png" data-base62-sha1="l0lFKANBuT7S4XsSwx1sRsBHfrr" width="448" height="367"></p>

**Reply by phexus23** [reply]
No, it is normal, happens to all people.

**Comment by rivermont** [comment]
<p>I had the same scenario, but when I confirmed and reloaded it now displays this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2a336a1ceb0f575350406482624d470bccfaeadb.png" data-download-href="/uploads/short-url/61kcWISh9r3zm3bdDcUtr31tXZx.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/25139386/181073226-87d48b13-b1df-49fc-b7be-b4c3b491d9a3.png" alt="image.png" data-base62-sha1="61kcWISh9r3zm3bdDcUtr31tXZx" width="690" height="275" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2a336a1ceb0f575350406482624d470bccfaeadb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">786×314 13.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So it looks like even though it says you will lose it, the site stays up. My site did not go down after I made the repo private. The message is quite confusing though.</p>

**Reply by amitmerchant1990** [reply]
Curious is it still working with the private repo?

**Comment by shornabho** [comment]
<p>Hey <a class="mention" href="/rivermont">@rivermont</a>, what Github means by that message is that, since the repository is private, people will not have access to the Github repository of the site code directly after you publish it with Github Pages. However, since the published site is public (can be accessed by anyone on the Internet) and Github only allows static sites, the code for the site will technically be available to the public, as anyone can view the Page Source of the published site to view the relevant code files and assets.</p>
<p>I hope this answers your doubt.</p>

**Reply by phatty88** [reply]
"the code for the site will technically be available to the public, as anyone can view the Page Source of the published site to view the relevant code files and assets."

would people be able to see api keys that were put in dotenv files even if you have a private account ?
sorry i am new to coding so im just trying to understand security a little more :(

**Reply by shornabho** [reply]
Hi @phatty88, 

As a general rule of thumb, consider any and every line of code you write in a frontend codebase always gets shipped to the client system for web interfaces in some way or form once you deploy it. Whether you have your API Keys in a dotenv or not, is just a coding best practice for larger environments to avoid code changes for multi-tenant or multi-environment deployments, such as you may want different deployment configs (API Keys, etc.) for production, development, staging environments. 

But most public APIs will have some security mechanism to ensure others cannot use the same API Key or Access Tokens such as domain-based filters on the backend. For example, your API backend might only choose to accept requests from `blog.example.com` and so if I steal your API Key and try to call the same API, since I don't own `blog.example.com` your backend API will not send me a valid response. Similarly for public APIs, or OAuth2 IDPs or OIDC partners they usually have a set of valid origins and valid redirect URIs that you can configure in the platform / app before you can use a particular set of access tokens. 

Hope this answers your question.

**Comment by Kanishka-Chandra** [comment]
You can explore netlify. Netlify allows to have a private repo and still host it as a website.
